crates/noirc_evaluator/src/ssa_refactor/ir/instruction_predicates.rs

```diff
@@ -0,0 +1,21 @@
+use super::{
+    super::basic_block::BasicBlockId, function::Function, instruction::TerminatorInstruction,
+};
+
+/// Visit all successors of a block with a given visitor closure. The closure
```

Comment:
The naming of this file is a bit odd to me. Is it intended to group together visitor functions? If so, perhaps we should rename it "visitors" or something more direct. "*_predicates" sounds like a bunch of boolean functions to me.